src/wallet/wallet.cpp

```diff
@@ -1970,37 +1970,38 @@ bool CWalletTx::InMempool() const
 
 bool CWalletTx::IsTrusted() const
 {
-    std::set<uint256> s;
-    return IsTrusted(s);
+    std::set<uint256> trusted_parents;
+    LOCK(pwallet->cs_wallet);
+    return pwallet->IsTrusted(*this, trusted_parents);
```

Comment:
772ea4844c34ad70d02fd0bd6c0945baa8fff85c
```suggestion
    AssertLockNotHeld(pwallet->cs_wallet);
    std::set<uint256> trusted_parents;
    LOCK(pwallet->cs_wallet);
    return pwallet->IsTrusted(*this, trusted_parents);
```